src/core/createEventManager.js

```diff
@@ -69,6 +58,14 @@ export default ({
           return consent.awaitConsent();
         })
         .then(() => {
+          // NOTE: this calls onBeforeEventSend callback (if configured)
+          event.finalize(onBeforeEventSend);
+
+          // if the callback returns false, the event should not be sent
+          if (!event.shouldSend()) {
+            return undefined;
+          }
+
           return sendEdgeNetworkRequest({
             request,
             runOnResponseCallbacks: onResponseCallbackAggregator.call,
```

Comment:
If there is an exception thrown in the onBeforeEventSend callback, it should call the onRequestFailure component callbacks. This is especially important for personalization. If the onBeforeEventSend callback throws an error, personalization needs to un-hide the containers.

Reply:
Additionally, if onBeforeEventSend callback returns false, the onRequestFailure component callbacks should be called.

Reply:
Ooooooh. This is very interesting and a good point. Let's talk about this one in today's check-in.